src/main/java/com/jagrosh/jmusicbot/settings/SettingsManager.java

```diff
@@ -40,14 +40,24 @@ public SettingsManager()
             JSONObject loadedSettings = new JSONObject(new String(Files.readAllBytes(Paths.get("serversettings.json"))));
             loadedSettings.keySet().forEach((id) -> {
                 JSONObject o = loadedSettings.getJSONObject(id);
+
+                // Legacy version support: On versions 0.2.7 and older, the repeat mode was represented as a boolean.
+                try
+                {
+                    if(o.getBoolean("repeat"))
+                        o.put("repeat", RepeatMode.On);
+                    else
+                        o.put("repeat", RepeatMode.Off);
+                } catch(JSONException e) { /* ignored */ }
```

Comment:
I'm not feeling comfortable about this bit. I can't really think of another solution without storing a "settings version" on the server settings.

Reply:
This seems alright. Another way to handle it might be to rename the repeat mode field from `repeat` to `repeat_mode` so that if `repeat` exists, we can always expect it to be a boolean.